src/components/CollapsingPortfolioTable/CollapsingPortfolioTable.tsx

```diff
@@ -0,0 +1,276 @@
+      setAssetSendQty("1");
+    }
+
+    setCollectTxDetails(true);
+  }, []);
+
+  const handleCopyAssetId = useCallback(
+    (toCopy: string) => {
+      navigator.clipboard.writeText(toCopy);
+      enqueueSnackbar(messageText.copy.success, { variant: "success" });
+    },
+    [enqueueSnackbar]
+  );
+
+  const RowsMemo = useMemo(() => {
+    if (rows === undefined) {
+      return <></>;
+    }
+
+    return rows.map((row: IPortfolioAssets) => <Row key={row.name} row={row} />);
+  }, [rows]);
+
+  // Keeps held assets up to date so table will be built properly
+  useEffect(() => {
+    const assetRows = heldAssets?.map((asset) => {
+      return createData(
+        asset.assetDetails.name,
+        asset.assetDetails.description,
+        asset.quantityQNT,
+        <AssetActionsStack
+          handleSendAsset={handleSendAsset}
+          handleCopyAssetId={handleCopyAssetId}
+          assetId={asset.asset}
+          assetName={asset.assetDetails.name}
+        />,
+        {
+          name: asset.assetDetails.name,
+          description: asset.assetDetails.description,
+          decimals: asset.assetDetails.decimals,
+          quantityQNT: `${asset.assetDetails.quantityQNT}`,
+        }
+      );
+    });
+
+    setRows(assetRows);
+  }, [blockHeight, handleCopyAssetId, handleSendAsset, heldAssets]);
+
+  return (
+    <TableContainer component={Paper}>
+      <Table aria-label="collapsible table">
+        <TableHead>
+          <TableRow>
+            <TableCell />
+            {/* Table Header row */}
+            {TopLevelHeaders.map((headerText, index) => {
+              return (
+                <TableCell align="center" key={`tc-${headerText}-${index}`}>
+                  {headerText}
+                </TableCell>
+              );
+            })}
+          </TableRow>
+        </TableHead>
+        <TableBody>
+          {/* Table Contents rows */}
+          {RowsMemo}
+        </TableBody>
+      </Table>
+      {collectTxDetails ? (
+        <JUPDialog isOpen={collectTxDetails} closeFn={handleClose}>
+          <Stack sx={{ alignItems: "center" }} spacing={2}>
+            <JUPInput inputType="address" placeholder='Enter "To" Address' fetchFn={(address) => fetchToAddress(address)}></JUPInput>
+            <JUPInput placeholder="Enter Quantity" inputType="quantity" fetchFn={(quantity) => fetchAssetQuantity(quantity)}></JUPInput>
+            <Button onClick={handleNext} variant="green">
+              Next
+            </Button>
+          </Stack>
+        </JUPDialog>
+      ) : (
+        <></>
+      )}
+    </TableContainer>
+  );
+};
+
+//
+// Helper functions
+//
+
+interface IAssetDetails {
+  name: string;
+  description: string;
+  decimals: number;
+  quantityQNT: string;
+}
+
+// creates row data based on inputs
+function createData(name: string, description: string, qtyOwned: string, assetActions: any, assetDetails: IAssetDetails) {
```

Comment:
FYI, on long params like this it's often a good idea to utilize an object with properties for each of these vars. reasoning is that the order of the properties of an object will not matter, whereas if there is an out of order param sent to this function, it may not get caught if the type still matches.

super small thing and imo, doesn't much matter. but i thought you'd prosper from the knowledge

Reply:
vs code had a "convert parameters to destructured object" refactor option which got me really close, so I used that and then wrote an interface to clean up the function definition

good stuff :)